lib/live_debugger_refactor/api/system/dbg.ex

```diff
@@ -0,0 +1,98 @@
+defmodule LiveDebugger.API.System.Dbg do
+  @moduledoc """
+  API for interacting with the Erlang's `:dbg` tracing functionalities.
+
+  ## Usage
+  1. If you want to trace callbacks you need to start tracer using `tracer/1` function.
+  Do not start it multiple times, as it will return an error if the tracer is already running.
+  2. Then using `process/1` you can enable tracing for all processes in the system.
+  3. To add a trace pattern for a specific function or module, use `trace_pattern/2`.
+  If you want to trace more information you can pass `match_spec` created with `flags_to_match_spec/1` function.
+  """
+
+  @tracing_flags [:return_trace, :exception_trace]
+
+  @type handler_spec() :: {handler_fun(), initial_data :: term()}
+  @type handler_fun() :: (event :: term(), data :: term() -> new_data :: term())
+
+  @callback tracer(handler :: handler_spec()) :: {:ok, pid()} | {:error, term()}
+  @callback process(flags :: list()) :: {:ok, term()} | {:error, term()}
+  @callback trace_pattern(module() | mfa(), match_spec :: term()) ::
+              {:ok, term()} | {:error, term()}
+  @callback clear_trace_pattern(module() | mfa()) :: {:ok, term()} | {:error, term()}
+
+  @doc """
+  Starts tracer process and returns its PID.
+  When tracer is already started, it returns error.
+  It uses `:dbg.tracer/2` under the hood.
+  """
+  def tracer(handler_spec), do: impl().tracer(handler_spec)
```

Comment:
I'd still add specs to these functions to make specs visible when you access its docs (by doing `h Module.function` or via VSCode extension)